contrib/kafka/kafkaeventsource-operator/pkg/controller/kafkaeventsource/kafkaeventsource_controller.go

```diff
@@ -0,0 +1,327 @@
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Create a new deployment for this EventSource
+	dep := deploymentForKafka(kafkaEventSource)
+
+	// Set KafkaEventSource as the owner and controller
+	if err := controllerutil.SetControllerReference(kafkaEventSource, dep, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if this Deployment already exists
+	found := &appsv1.Deployment{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: dep.Name, Namespace: dep.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		//deployment not found
+		log.Printf("Creating a new Deployment %s/%s\n", dep.Namespace, dep.Name)
+		err = r.client.Create(context.TODO(), dep)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Deployment created successfully - requeue
+		return reconcile.Result{Requeue: true}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check the config - Environment Variables and annotations
+	if !reflect.DeepEqual(dep.Spec.Template, found.Spec.Template) {
+		found.Spec.Template = dep.Spec.Template
+		err := r.client.Update(context.TODO(), found)
+		if err != nil {
+			log.Printf("failed to update Deployment configuration:, %s", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Ensure the deployment size is the same as the spec
+	size := kafkaEventSource.Spec.Replicas
+	if size == nil {
+		intendedSize := int32(1)
+		size = &intendedSize
+	}
+
+	if *found.Spec.Replicas != *size {
+		*found.Spec.Replicas = *size
+		log.Printf("Setting replicas:%d", *size)
+		err = r.client.Update(context.TODO(), found)
+		if err != nil {
+			log.Printf("Failed to update Deployment: %s", found.Name)
+			return reconcile.Result{}, err
+		}
+		// Spec updated - return and requeue
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Update the KafkaEventSource status with the pod names
+	// List the pods for this event source's deployment
+	podList := &corev1.PodList{}
+	labelSelector := labels.SelectorFromSet(labelsForKafkaEventSource(kafkaEventSource.Name))
+	listOps := &client.ListOptions{Namespace: kafkaEventSource.Namespace, LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), listOps, podList)
+	if err != nil {
+		log.Printf("Failed to list pods when reconciling KafkaEventSource: %s", kafkaEventSource.Name)
+		return reconcile.Result{}, err
+	}
+
+	podNames := getPodNames(podList.Items)
+
+	// Update status.Nodes if needed
+	if !reflect.DeepEqual(podNames, kafkaEventSource.Status.Nodes) {
+		kafkaEventSource.Status.Nodes = podNames
+		log.Printf("Updating EventSource.Status.Nodes")
+		err := r.client.Update(context.TODO(), kafkaEventSource)
+		if err != nil {
+			log.Printf("failed to update KafkaEventSource status:, %s", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Deployment already exists don't requeue
+	log.Printf("Skip reconcile: Deployment %s/%s already exists", found.Namespace, found.Name)
```

Comment:
This seems like an inaccurate statement, the `Deployment` may have been altered due to reconciliation.